thetis/solver2d.py

```diff
@@ -374,28 +384,22 @@ def create_equations(self):
         self.equations.sw.bnd_functions = self.bnd_functions['shallow_water']
         uv_2d, elev_2d = self.fields.solution_2d.split()
         if self.options.solve_tracer:
-            if self.options.tracer_metadata == {}:
-                md = field_metadata['tracer_2d'].copy()
-                md['source'] = self.options.tracer_source_2d
-                self.options.tracer_metadata['tracer_2d'] = md
-                self.add_new_field(Function(self.function_spaces.Q_2d, name='tracer_2d'),
-                                   'tracer_2d',
-                                   md['name'],
-                                   md['filename'],
-                                   shortname=md['shortname'],
-                                   unit=md['unit'])
+            if self.options.tracer == {}:
+                self.options.add_tracer_2d('tracer_2d', 'Depth averaged tracer', 'Tracer2d', 'Tracer', '-',
+                                           source=self.options.tracer_source_2d,
+                                           diffusivity=self.options.horizontal_diffusivity)
```

Comment:
I guess this is for backwards compatibility. Should we deprecate this code path in the future? That is, use always has to define tracers via `options.add_tracer_2d`? That would imply removing `options.solve_tracer`.

Reply:
Sure. We would then also be able to drop `options.horizontal_diffusivity` and `options.tracer_source_2d`. I'll add in a deprecation warning.